server/app/forms/QuestionForm.java

```diff
@@ -192,4 +198,46 @@ public final boolean isUniversal() {
   public final void setIsUniversal(boolean universal) {
     this.isUniversal = universal;
   }
+
+  public final ImmutableSet<PrimaryApplicantInfoTag> primaryApplicantInfoTags() {
+    return this.primaryApplicantInfoTags;
+  }
+
+  // The names of these functions must match up to the fieldName value in the
+  // PrimaryApplicantInfoTag
+  public final void setPrimaryApplicantDob(boolean val) {
+    setTagState(PrimaryApplicantInfoTag.APPLICANT_DOB, val);
+  }
+
+  public final void setPrimaryApplicantEmail(boolean val) {
+    setTagState(PrimaryApplicantInfoTag.APPLICANT_EMAIL, val);
+  }
+
+  public final void setPrimaryApplicantName(boolean val) {
+    setTagState(PrimaryApplicantInfoTag.APPLICANT_NAME, val);
+  }
+
+  public final void setPrimaryApplicantPhone(boolean val) {
+    setTagState(PrimaryApplicantInfoTag.APPLICANT_PHONE, val);
+  }
+
+  private void setTagState(PrimaryApplicantInfoTag primaryApplicantInfoTag, boolean isSet) {
+    boolean present = this.primaryApplicantInfoTags.contains(primaryApplicantInfoTag);
+    if (isSet) {
+      this.primaryApplicantInfoTags =
+          present
+              ? this.primaryApplicantInfoTags
+              : new ImmutableSet.Builder<PrimaryApplicantInfoTag>()
+                  .addAll(this.primaryApplicantInfoTags)
+                  .add(primaryApplicantInfoTag)
+                  .build();
+    } else {
+      this.primaryApplicantInfoTags =
+          present
+              ? this.primaryApplicantInfoTags().stream()
+                  .filter(t -> t != primaryApplicantInfoTag)
+                  .collect(ImmutableSet.toImmutableSet())
+              : this.primaryApplicantInfoTags;
+    }
```

Comment:
```suggestion
    boolean present = this.primaryApplicantInfoTags.contains(primaryApplicantInfoTag);

    if (isSet && !present) {
      this.primaryApplicantInfoTags = new ImmutableSet.Builder<PrimaryApplicantInfoTag>()
                  .addAll(this.primaryApplicantInfoTags)
                  .add(primaryApplicantInfoTag)
                  .build();
    } else if (!isSet && present) {
      this.primaryApplicantInfoTags = this.primaryApplicantInfoTags().stream()
                  .filter(t -> t != primaryApplicantInfoTag)
                  .collect(ImmutableSet.toImmutableSet());
    }
```

I don't think we need the nested ternarys. Double check this is functionally equivalent.